docs/diagrams/ResidentRoomAllocationDiagram.puml

```diff
@@ -0,0 +1,23 @@
+@startuml
+'https://plantuml.com/activity-diagram-beta
+
+start
+:Check if Room exists;
+if () then ([Room exists])
+    :Check if Resident exists;
+    if () then ([Resident exists])
+        : Check if Resident or Room is already allocated;
+        if () then ([None allocated])
+            : Set the room number in Resident;
+            : Set the occupancy as 'Y' in Room;
+            stop
+        else ([Resident or/and Room is allocated])
+            stop
+    endif
+    else ([Resident does not exist])
```

Comment:
For consistency with other activity diagrams, perhaps state the actual condition in the "happy flow" and "else" for the flow that terminates.